entity-framework/core/miscellaneous/spatial.md

```diff
@@ -0,0 +1,295 @@
+
+EF Core Provider                        | Spatial NuGet Package
+--------------------------------------- | ---------------------
+Microsoft.EntityFrameworkCore.SqlServer | [Microsoft.EntityFrameworkCore.SqlServer.NetTopologySuite](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer.NetTopologySuite)
+Microsoft.EntityFrameworkCore.Sqlite    | [Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite)
+Microsoft.EntityFrameworkCore.InMemory  | [NetTopologySuite](https://www.nuget.org/packages/NetTopologySuite)
+Npgsql.EntityFrameworkCore.PostgreSQL   | [Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite](https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite)
+
+## Reverse engineering
+
+The spatial NuGet packages also enable [reverse engineering](../managing-schemas/scaffolding.md) models with spatial properties, but you need to install the package ***before*** running `Scaffold-DbContext` or `dotnet ef dbcontext scaffold`. If you don't, you'll receive warnings about not finding type mappings for the columns and the columns will be skipped.
+
+## NetTopologySuite (NTS)
+
+NetTopologySuite is a spatial library for .NET. EF Core enables mapping to spatial data types in the database by using NTS types in your model.
+
+To enable mapping to spatial types via NTS, call the UseNetTopologySuite method on the provider's DbContext options builder. For example, with SQL Server you'd call it like this.
+
+``` csharp
+optionsBuilder.UseSqlServer(
+    @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=WideWorldImporters",
+    x => x.UseNetTopologySuite());
+```
+
+There are several spatial data types. Which type you use depends on the types of shapes you want to allow. Here is the hierarchy of NTS types that you can use for properties in your model. They're located within the `NetTopologySuite.Geometries` namespace. Corresponding interfaces in the GeoAPI package can also be used.
+
+* Geometry
+  * Point
+  * LineString
+  * Polygon
+  * GeometryCollection
+    * MultiPoint
+    * MultiLineString
+    * MultiPolygon
+
+> [!WARNING]
+> CircularString, CompoundCurve, and CurePolygon aren't supported by NTS.
+
+The following entity classes could be used to map to tables in the [Wide World Importers sample database](http://go.microsoft.com/fwlink/?LinkID=800630).
+
+``` csharp
+[Table("Cities", Schema = "Application"))]
+class City
+{
+    public int CityID { get; set; }
+
+    public string CityName { get; set; }
+
+    public IPoint Location { get; set; }
+}
+
+[Table("Countries", Schema = "Application"))]
+class Country
+{
+    public int CountryID { get; set; }
+
+    public string CountryName { get; set; }
+
+    // Database includes both Polygon and MultiPolygon values
+    public IGeometry Border { get; set; }
```

Comment:
The comment is good. But I wonder if there is worth discussing in the text when it makes sense to constrain a property type (e.g. if all the instances are going to be a Point) vs. leaving it as Geometry, after showing this example.